pkg/repositories/repositories.go

```diff
@@ -57,6 +57,44 @@ func NewConfig(filename string) (Config, error) {
 	return repoConfig, nil
 }
 
+// If our first config and our second config share a repository,
+// we set the afterVersion of the first to its original version, and the
+// version of the first to the version of the second.
+func CombineConfigs(currentConfig *Config, newConfig *Config) {
+
+	// Extract new repos regardless of category
+	var newRepos []Repository
+	for _, category := range newConfig.Section.Categories {
+		newRepos = append(newRepos, category.Repos...)
+	}
+
+	// Iterate through current categories and repositories, maintaining
+	// the structure.
+	for categoryIdx := range currentConfig.Section.Categories {
+		for repoIdx, currentRepo := range currentConfig.Section.Categories[categoryIdx].
+			Repos {
+			// If two repo structs share a url, they are assumed to be the same repo.
+			// Store the newVersion as the version for the current repo.
+			if newVersion := contains(newRepos, currentRepo); newVersion != "" {
+				remappedRepo 			 := currentRepo
+				remappedRepo.AfterVersion = currentRepo.Version
+				remappedRepo.Version	  = newVersion
+
+				currentConfig.Section.Categories[categoryIdx].Repos[repoIdx] = remappedRepo
+			}
+		}
+	}
+}
+
+func contains(r []Repository, e Repository) string {
+	for _, a := range r {
```

Comment:
This line and the line above have some really unintuitive var names (`a` and `e`). `r` should probably also be renamed since "r" implies a single object, not an array of objects.

Reply:
Totally right, I planned to give this more descriptive names once I was sure this was the right approach